In [134]:
import numpy as np
import pandas as pd

In [135]:
dataDir = '/home/luke/Documents/MINOS/Minos/data/'

baseFile = dataDir + 'final_US/2018_US_cohort.csv'

In [136]:
dat = pd.read_csv(baseFile)

In [137]:
len(dat)

28191

In [138]:
new_wave = dat[(dat['age'] == 16)]
len(new_wave)

272

In [139]:
for a in range(0,110):
    print(f"Age: {a}, Count: {len(dat[(dat['age'] == a)])}")


Age: 0, Count: 0
Age: 1, Count: 0
Age: 2, Count: 0
Age: 3, Count: 0
Age: 4, Count: 0
Age: 5, Count: 0
Age: 6, Count: 0
Age: 7, Count: 0
Age: 8, Count: 0
Age: 9, Count: 0
Age: 10, Count: 0
Age: 11, Count: 0
Age: 12, Count: 0
Age: 13, Count: 0
Age: 14, Count: 0
Age: 15, Count: 0
Age: 16, Count: 272
Age: 17, Count: 451
Age: 18, Count: 439
Age: 19, Count: 442
Age: 20, Count: 393
Age: 21, Count: 424
Age: 22, Count: 381
Age: 23, Count: 375
Age: 24, Count: 294
Age: 25, Count: 419
Age: 26, Count: 392
Age: 27, Count: 328
Age: 28, Count: 364
Age: 29, Count: 329
Age: 30, Count: 348
Age: 31, Count: 386
Age: 32, Count: 326
Age: 33, Count: 371
Age: 34, Count: 393
Age: 35, Count: 393
Age: 36, Count: 415
Age: 37, Count: 410
Age: 38, Count: 488
Age: 39, Count: 458
Age: 40, Count: 432
Age: 41, Count: 471
Age: 42, Count: 422
Age: 43, Count: 449
Age: 44, Count: 500
Age: 45, Count: 501
Age: 46, Count: 499
Age: 47, Count: 503
Age: 48, Count: 556
Age: 49, Count: 519
Age: 50, Count: 555
Age: 51, Count: 554
Ag

Testing above shows that the 16 year old group has 272 members, whilst the range of counts for each age group is generally between 250-500 for the 16-~80 year olds. This could be an issue, as we will be shrinking our population over time. To combat this, we could take the 16 year olds from 2 waves of the survey (i.e. from 2018 & 2019) and call this a single cohort.

Now lets check the outputs to see how many of each age group we have, and whether there is a reduction in the overall sample.

In [140]:
# Run 1 (this one) is the original with no modification to the replenishing population

outDir = '/home/luke/Documents/MINOS/Minos/output/'
run = '2022_08_16_14_12_53/'

dat1 = pd.read_csv(outDir + run + 'simulation_data/' + '2012.csv')
dat2 = pd.read_csv(outDir + run + 'simulation_data/' + '2013.csv')
dat3 = pd.read_csv(outDir + run + 'simulation_data/' + '2014.csv')
dat4 = pd.read_csv(outDir + run + 'simulation_data/' + '2015.csv')
dat5 = pd.read_csv(outDir + run + 'simulation_data/' + '2016.csv')

#dat1.columns

In [141]:
print(len(dat1[(dat1['alive'] == 'alive')]))
print(len(dat2[(dat2['alive'] == 'alive')]))
print(len(dat3[(dat3['alive'] == 'alive')]))
print(len(dat4[(dat4['alive'] == 'alive')]))
print(len(dat5[(dat5['alive'] == 'alive')]))

print(len(dat1[(dat1['age'] == 16)]))
print(len(dat2[(dat2['age'] == 16)]))
print(len(dat3[(dat3['age'] == 16)]))
print(len(dat4[(dat4['age'] == 16)]))
print(len(dat5[(dat5['age'] == 16)]))

print(dat5['age'].value_counts())

38847
38475
38132
37699
37322
1
1
1
1
1
23.0     805
49.0     804
46.0     802
47.0     800
48.0     798
        ... 
19.0       2
100.0      2
101.0      2
103.0      1
16.0       1
Name: age, Length: 84, dtype: int64


In [142]:
run2 = '2022_08_16_14_17_10/'

dat11 = pd.read_csv(outDir + run2 + 'simulation_data/' + '2012.csv')
dat12 = pd.read_csv(outDir + run2 + 'simulation_data/' + '2013.csv')
dat13 = pd.read_csv(outDir + run2 + 'simulation_data/' + '2014.csv')
dat14 = pd.read_csv(outDir + run2 + 'simulation_data/' + '2015.csv')
dat15 = pd.read_csv(outDir + run2 + 'simulation_data/' + '2016.csv')

print(len(dat11[(dat11['alive'] == 'alive')]))
print(len(dat12[(dat12['alive'] == 'alive')]))
print(len(dat13[(dat13['alive'] == 'alive')]))
print(len(dat14[(dat14['alive'] == 'alive')]))
print(len(dat15[(dat15['alive'] == 'alive')]))

print(len(dat11[(dat11['age'] == 16)]))
print(len(dat12[(dat12['age'] == 16)]))
print(len(dat13[(dat13['age'] == 16)]))
print(len(dat14[(dat14['age'] == 16)]))
print(len(dat15[(dat15['age'] == 16)]))

print(dat15['age'].value_counts())

38875
38543
38161
37752
37373
0
0
0
0
0
23.0     809
47.0     809
46.0     804
48.0     799
49.0     799
        ... 
20.0       1
104.0      1
102.0      1
18.0       1
19.0       1
Name: age, Length: 87, dtype: int64


So there seems to be no 16 year olds in either of the runs, with our without the replenishment change. Need to test the input data files and see if we're removing them elsewhere or something.

In [143]:
rawFile = dataDir + 'raw_US/2018_US_cohort.csv'

rawDat = pd.read_csv(rawFile)

In [144]:
print(len(rawDat[(rawDat['age'] == 16)]))

404


Definitely do have 16 year olds in the raw files then, must be removing them somewhere. Need to check the data generation pipeline.

In [145]:
correctedFile = dataDir + 'corrected_US/2018_US_cohort.csv'
correctedDat = pd.read_csv(correctedFile)
print(len(correctedDat[(correctedDat['age'] == 16)]))

300


In [146]:
compositeFile = dataDir + 'composite_US/2018_US_cohort.csv'
compositeDat = pd.read_csv(compositeFile)
print(len(compositeDat[(compositeDat['age'] == 16)]))

300


In [147]:
finalFile = dataDir + 'final_US/2018_US_cohort.csv'
finalDat = pd.read_csv(finalFile)
print(len(finalDat[(finalDat['age'] == 16)]))

final_ages = finalDat['age'].value_counts()
final_ages = final_ages.sort_index()

272


We have identified the problem!! Every dataset apart from the final_US data includes 16 year olds, and so the step that must be removing the younger ages is the complete case in `US_complete_case.py`. There are a couple of possible reasons for missing data across all ages:
- Missing labour state
- Missing education state
- Some information (such as ethnicity) is recorded in the youth dataset and not filled in for anyone under that age of 18 (as 19 is the youngest age seen in the

In [148]:
repl = pd.read_csv(dataDir + 'replenishing/replenishing_pop_2019-2070.csv')

In [149]:
# 216976903
# 69199241

person1 = 69199241
person1_2 = person1 + 2029 + 1000000

record1 = repl[(repl['pidp'] == person1)]
record1_2 = repl[(repl['pidp'] == person1_2)]

print(record1['weight'])
print(record1_2['weight'])


0    2.000858
Name: weight, dtype: float64
Series([], Name: weight, dtype: float64)


In [150]:
# test the reweight here
repl = pd.read_csv(dataDir + 'replenishing/replenishing_pop_2019-2070.csv')
summed_weights = repl.groupby(['sex', 'time'])['weight'].sum().reset_index()
summed_weights = summed_weights.rename(columns = {'weight':'sum_weight', 'year':'time'})
projections = pd.read_csv('/home/luke/Documents/MINOS/Minos/persistent_data/pop_projections_2008-2070.csv')
projections = projections.drop(labels = 'Unnamed: 0', axis=1)
projections = projections.rename(columns = {'year':'time'})

new_repl = repl

# now reweight new population file
#for sex in ['Male', 'Female']:
#    for year in range(2019, 2071, 1):
#        new_repl['weight'][(new_repl['sex'] == sex) & (new_repl['time'] == year)] = (new_repl['weight'] * projections['count'][(projections['sex'] == sex) & projections['year'] == year]) / summed_weights['weight'][(summed_weights['sex'] == sex) & (summed_weights['time'] == year)]

new_repl['new_weight'] = pd.NA
new_repl = new_repl.merge(projections, on = ['time', 'sex', 'age'])
new_repl = new_repl.merge(summed_weights, on = ['time', 'sex'])


#for sex in ['Male', 'Female']:
#    for year in range(2019, 2071, 1):
new_repl['new_weight'] = (new_repl['weight'] * new_repl['count']) / new_repl['sum_weight']


In [152]:
print(repl['weight'][(repl['pidp'] == 71820040)])
print(new_repl['weight'][(new_repl['pidp'] == 71820040)])
print(new_repl['new_weight'][(new_repl['pidp'] == 71820040)])

551    0.964942
Name: weight, dtype: float64
546    0.964942
Name: weight, dtype: float64
546    4266.878148
Name: new_weight, dtype: float64
